vignettes/ABN_inference.Rmd

```diff
@@ -15,201 +17,324 @@ knitr::opts_chunk$set(
+
+where $\mathscr{C}_j$ denotes the set of parents of $B_j$. 
+
+
+## Inference with partial information
+
+In this section, we tackle the more general case, where we do not have access to the full Markov blanket of a node. To estimate the conditional distribution of such a node, we go through the graph iteratively. We begin by estimating the values of the nodes using upstream information (from the parents), and then, we refine these estimates by incorporating downstream information (from the children and the children's parents), as described in Section 2. We thus propagate the estimates throughout the network to refine our understanding of the node's distribution, given the partial information available.
+
+We proceeed with the following steps: 
+
+1. We first order the nodes of the graph in a topological order, ensuring that each node appears before all the nodes it is connected to.
+
+
+2. Following the topological order, we iteratively estimate the distribution of each node based only on its parents (as described in Step 1).
+
+For nodes with no parents (i.e. root nodes in the graph), the distribution is directly estimated using the observed data:
+
+  - observed mean for Poisson nodes, 
   
-  - if $Y$ is Poisson, $Y \sim P(\lambda_Y)$ with $\lambda_Y = \exp \left( \beta_{Y0} + \beta_{Y1} X \right)$.
+  - observed mean and variance for Gaussian nodes,
   
-Now, depending on the cases:
+  - observed frequencies for Binomial nodes.
+
+For the remaining nodes, we use the estimated distributions of their parents as values in the formula, unless those parent nodes are evidence (i.e. observed). Specifically:
+
+  - if a parent $A$ is Gaussian, we use its estimated expectation as its value, 
+
+  - if a parent $A$ is Poisson, we use its estimated expectation as its value,
+
+  - if a parent $A$ is Binomial, we marginalize over its possible values.
+
+3. Following the reverse topological order (from leaves to roots), we iteratively update the distribution of each node based on its children and their parents (as described in Step 2) until reaching the node of interest.
+
+For nodes with no children (i.e. leaf nodes in the graph), the distribution remains unchanged.
+
+For the remaining nodes, we use the estimated distributions of their children's parents as values in the formula, unless those parent nodes are evidence (i.e. observed). Specifically:
+
+  - if a parents $C$ is Gaussian, we use its estimated expectation as its value,
+
+  - if a parent $C$ is Poisson, we use its estimated expectation as its value,
+
+  - if a parent $C$ is Binomial, we marginalize over its possible values. 
+
+
+*Note*: The multinomial case is not yet implemented in the current version of the `PredictABN` function. However, if it were to be included, the procedure would remain the same. In this case, we would treat multinomial nodes as binomial nodes, but we would need to marginalize over all possible values of the multinomial distribution.
```

Comment:
If I understand this correctly, it would require a simple 0-1 encoding step before applying the Binomial procedure to the individual levels of the multinomial variables. Here, we should spend some thought about encoding all (N) variables or (N-1) and also about how to combine this back into a multinomial variable.